oauth2_http/java/com/google/auth/oauth2/UserCredentials.java

```diff
@@ -244,6 +249,45 @@ public final String getRefreshToken() {
     return refreshToken;
   }
 
+
+  /**
+   * Returns the instance of InputStream containing the following user credentials in JSON format:
+   *  - RefreshToken
+   *  - ClientId
+   *  - ClientSecret
+   *  - ServerTokenUri
+   *
+   * @return user credentials stream
+   */
+  private InputStream getUserCredentialsStream() throws IOException {
+    GenericJson json = new GenericJson();
+    json.put("type", GoogleCredentials.USER_FILE_TYPE);
+    if (refreshToken != null) {
+      json.put("refresh_token", refreshToken);
+    }
+    if (tokenServerUri != null) {
+      json.put("token_server_uri", tokenServerUri);
+    }
+    if (clientId != null) {
+      json.put("client_id", clientId);
+    }
+    if (clientSecret != null) {
+      json.put("client_secret", clientSecret);
+    }
+    json.setFactory(JSON_FACTORY);
+    String text = json.toPrettyString();
+    return new ByteArrayInputStream(text.getBytes(UTF_8));
+  }
+
+  /**
+   * Saves the end user credentials in file
+   *
+   * @throws IOException An error storing the credentials.
+   */
+  public void saveUserCredentials() throws IOException {
+    super.saveCredentialsToFile(getUserCredentialsStream(), USER_CREDENTIALS_FILE_NAME);
```

Comment:
I don't feel comfortable hardcoding the path where this is saved.

Also, the "UserCredentials" part of this name is redundant since it lives in the UserCredentials class.